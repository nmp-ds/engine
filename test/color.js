import { setup } from "./_helpers.js";
import { expect, test } from "vitest";
import { opacity } from '#theme';


setup();

test('opacity by theme', async ({ uno }) => {
  const classes = Object.keys(opacity).map((op => `opacity-${op}`));

  const { css } = await uno.generate(classes);
  expect(css).toMatchSnapshot();
});

test('opacity not created if invalid', async ({ uno }) => {
  const classes = ['opacity-1'];

  const { css } = await uno.generate(classes);
  expect(css).toMatchInlineSnapshot('""');
});

test('text colors', async({ uno }) => {
  const classes = ['text' ,'text-inverted', 'text-inverted-subtle', 'text-subtle'];

  const { css } = await uno.generate(classes);
  expect(css).toMatchSnapshot();
});

test('text color invalid class', async({ uno }) => {
  const classes = ['text-color'];

  const { css } = await uno.generate(classes);
  expect(css).toMatchInlineSnapshot('""');
});

test('bg colors', async({ uno }) => {
  const classes = [
    'bg',
    'bg-subtle',
    'bg-inherit',
    'bg-transparent',
    'bg-current'];

  const { css } = await uno.generate(classes);
  expect(css).toMatchSnapshot();
});

test('bg color invalid class', async({ uno }) => {
  const classes = ['bg-color'];

  const { css } = await uno.generate(classes);
  expect(css).toMatchInlineSnapshot('""');
});

test('caret', async({ uno }) => {
  const classes = [
    'caret',
    'caret-inherit',
    'caret-current',
    'caret-transparent',
  ];

  const { css } = await uno.generate(classes);
  expect(css).toMatchSnapshot();
});