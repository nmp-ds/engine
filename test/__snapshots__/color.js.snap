// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`bg colors 1`] = `
"/* layer: default */
.bg{background-color:var(--w-color-background);}
.bg-subtle{background-color:var(--w-color-background-subtle);}
.bg-inherit{background-color:inherit;}
.bg-transparent{background-color:transparent;}
.bg-current{background-color:currentColor;}"
`;

exports[`caret 1`] = `
"/* layer: default */
.caret-inherit{caret-color:inherit;}
.caret-current{caret-color:currentColor;}
.caret-transparent{caret-color:transparent;}"
`;

exports[`opacity by theme 1`] = `
"/* layer: default */
.opacity-0{opacity:0%;}
.opacity-100{opacity:100%;}
.opacity-25{opacity:25%;}
.opacity-50{opacity:50%;}"
`;

exports[`text colors 1`] = `
"/* layer: default */
.text{color:var(--w-color-text);}
.text-inverted{color:var(--w-color-text-inverted);}
.text-inverted-subtle{color:var(--w-color-text-inverted-subtle);}
.text-subtle{color:var(--w-color-text-subtle);}"
`;
